confgenerator/confgenerator.go

```diff
@@ -286,6 +287,17 @@ func (uc *UnifiedConfig) GenerateFluentBitConfigs(logsDir string, stateDir strin
 			return "", "", err
 		}
 	}
+
+	// make sure all collections are sorted so that generated configs are consistently generated
+	sort.Slice(fbTails, func(i, j int) bool { return fbTails[i].Tag < fbTails[j].Tag })
```

Comment:
Might worth a shared method for the repeated logic here? The only caveat seems to be `fbFilterParserGroups` that has slightly different logic.

Reply:
I suspect these will go away altogether in a future commit, because we can generate these in source config order.